hub/api/tests/test_update_samples.py

```diff
@@ -252,3 +248,35 @@ def test_warnings(memory_ds):
     # this update makes (large) suboptimal chunks
     with pytest.warns(UserWarning):
         tensor[:] = np.zeros((10, 32, 31), dtype="int32")
+
+
+def test_inplace_updates(memory_ds):
+    ds = memory_ds
+    ds.create_tensor("x")
+    ds.x.extend(np.zeros((32, 32)))
+    ds.x += 1
+    np.testing.assert_array_equal(ds.x.numpy(), np.ones((32, 32)))
+    ds.x += ds.x
+    np.testing.assert_array_equal(ds.x.numpy(), np.ones((32, 32)) * 2)
+    ds.x *= np.zeros((32,))
+    np.testing.assert_array_equal(ds.x.numpy(), np.zeros((32, 32)))
+    ds.x -= 4
+    ds.x /= 2
+    np.testing.assert_array_equal(ds.x.numpy(), -np.ones((32, 32)) * 4 / 2)
+    ds.x[:16] *= 0
+    np.testing.assert_array_equal(
+        ds.x.numpy(), np.concatenate([np.zeros((16, 32)), -np.ones((16, 32)) * 4 / 2])
+    )
+
+    compressions = [
+        {"sample_compression": "jpeg"},
+        {"chunk_compression": "jpeg"},
+        {"sample_compression": "png"},
+        {"chunk_compression": "png"},
+    ]
```

Comment:
we should probably create a general parametrize for all these

Reply:
Done.